### ISSUE TITLE
Deprecation warnings about upb

### ISSUE THREAD

**Issue by alokpr** [grpc/grpc#35431]
### What version of gRPC and what language are you using?
1.60.0, C++

### What operating system (Linux, Windows,...) and version?
Linux

### What did you do?
bazel build //...

### What did you expect to see?
No warnings

### What did you see instead?
WARNING: WORKSPACE: in bind rule //external:upb_lib: target '//external:upb_lib' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead

This is being reported from here - https://github.com/protocolbuffers/protobuf/blob/6b5d8db01fe47478e8d400f550e797e6230d464e/upb/BUILD#L95

**Comment by rdesgroppes** [comment]
Same here:
```
WARNING: /path/to/WORKSPACE:207:10: in bind rule //external:upb_lib: target '//external:upb_lib' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/BUILD:1857:16: in cc_library rule @com_github_grpc_grpc//:alts_util: target '@com_github_grpc_grpc//:alts_util' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/src/core/BUILD:282:16: in cc_library rule @com_github_grpc_grpc//src/core:status_helper: target '@com_github_grpc_grpc//src/core:status_helper' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/src/core/BUILD:5397:16: in cc_library rule @com_github_grpc_grpc//src/core:grpc_backend_metric_filter: target '@com_github_grpc_grpc//src/core:grpc_backend_metric_filter' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/BUILD:3656:16: in cc_library rule @com_github_grpc_grpc//:xds_client: target '@com_github_grpc_grpc//:xds_client' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/BUILD:3426:16: in cc_library rule @com_github_grpc_grpc//:tsi_alts_credentials: target '@com_github_grpc_grpc//:tsi_alts_credentials' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/BUILD:3006:16: in cc_library rule @com_github_grpc_grpc//:grpc_client_channel: target '@com_github_grpc_grpc//:grpc_client_channel' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/src/core/BUILD:4106:16: in cc_library rule @com_github_grpc_grpc//src/core:grpc_lb_policy_grpclb: target '@com_github_grpc_grpc//src/core:grpc_lb_policy_grpclb' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/src/core/BUILD:4834:16: in cc_library rule @com_github_grpc_grpc//src/core:health_check_client: target '@com_github_grpc_grpc//src/core:health_check_client' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/src/core/BUILD:4223:16: in cc_library rule @com_github_grpc_grpc//src/core:grpc_lb_policy_rls: target '@com_github_grpc_grpc//src/core:grpc_lb_policy_rls' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/src/core/BUILD:4302:16: in cc_library rule @com_github_grpc_grpc//src/core:grpc_xds_client: target '@com_github_grpc_grpc//src/core:grpc_xds_client' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
WARNING: /path/to/bazel/cache/external/com_github_grpc_grpc/BUILD:1906:16: in cc_library rule @com_github_grpc_grpc//:grpc++_base: target '@com_github_grpc_grpc//:grpc++_base' depends on deprecated target '@com_google_protobuf//upb:upb': use upb:base and/or upb:mem instead
```

**Comment by yashykt** [comment]
cc @ericsalo

**Comment by ericsalo** [comment]
Yes. gRPC needs to be refactored to stop using these old upb artifacts. In fact I am working on a patch for this now but progress is relatively slow as I do not understand most of the gRPC build details.

**Comment by ericsalo** [comment]
https://github.com/grpc/grpc/pull/35506 is now in the pipeline.

**Comment by veblush** [comment]
As Eric mentioned above, this is fixed on the master and will be available from gRPC 1.61 or later.
